### PR TITLE
Use a task to download the image payload

### PR DESCRIPTION
**Pull request by poncovka**
* Use the task in the image payload class.
* Add unit tests for the image download.

### CODE REVIEW

File: pyanaconda/modules/payloads/payload/live_image/download_progress.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        """Report the progress."""
+        pct = min(int(100 * self._downloaded_size / self._total_size), 100)
+
+        if pct == self._last_pct:
+            return
+
+        self._last_pct = pct
+
+        log.debug("Downloaded %s (%s%%)", Size(self._downloaded_size), pct)
+        self._callback(_("Downloading {} ({}%)").format(self._url, pct))
```
Comment (VladimirSlavik) [inline pyanaconda/modules/payloads/payload/live_image/download_progress.py:54]: What do you think about unifying this? I find it fascinating that the debug call does not mention the URL, while the callback does not include the size, so you need both to get full information.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

File: pyanaconda/modules/payloads/payload/live_image/download_progress.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/payloads/payload/live_image/download_progress.py:54]: I don't think we want to do that. The size woudn't say anything useful to the user and the URL wouldn't say anything useful to us. We would just burden everyone with unnecessary data.

**Pull request merged**